watcher.go

```diff
@@ -199,8 +199,12 @@ func (w *Watcher) watch() {
 // AddPin adds a new pin to be watched for changes
 // The pin provided should be the pin known by the kernel
 func (w *Watcher) AddPin(p uint) {
-	pin := NewInput(p)
-	setEdgeTrigger(pin, edgeBoth)
+	w.AddPinWithEdgeAndLogic(p, ActiveHigh, EdgeBoth)
+}
+
+func (w *Watcher) AddPinWithEdgeAndLogic(p uint, logicLevel LogicLevel, edge Edge) {
```

Comment:
The function name order should mirror the order of arguments. Here level and edge are reversed.